galley/pkg/server/args.go

```diff
@@ -92,6 +92,9 @@ type Args struct {
 	// Enable galley server mode
 	EnableServer bool
 
+	// Enable service discovery / endpoint processing.
+	EnableServiceDiscovery bool
```

Comment:
Why is the flag called something different here?

Reply:
EnableServiceDiscovery is a general scoped name, which implies listening to additional resources etc. The other name, SynthesizeServiceEntries, is specific to the processor, which wires up the synthesizer, as part of enabling service discovery.

Reply:
I don't see where this flag controls which resources we're listening to. It only seems to control the value of `SynthesizeServiceEntries`.

Reply:
It is used down below, where we calculate the set of input/output resources. :)